Cod3rsGrowth.web/wwwroot/app/anime/CadastroEditarAnime.controller.js

```diff
@@ -19,34 +22,75 @@ sap.ui.define([
 	const VALOR_INICIAL_DA_NOTA = 1;
 	const POSICAO_INICIAL_DA_LISTA = 0;
 	const POSICAO_DO_ID = 1;
-	const FALHA_NA_REQUISIÇÃO = "Ocorreu um ou mais erros na requisição";
 	const VALUE_STATE_ERROR = "Error";
 	const VALUE_STATE_NONE = "None";
 	const VALUE_STATE_NOME_OBRIGATORIO = "o campo nome é obrigatório";
 	const VALUE_STATE_SINOPSE_OBRIGATORIA = "o campo sinopse é obrigatório";
 	const VALUE_STATE_DATA_OBRIGATORIO = "o campo data lançamento é obrigatório";
-	const DATA_PADRAO_INICIAL = "24/07/2024";
 	const MESSAGEM_PRECISA_SELECIONAR_GENERO = "Precisa selecionar ao menos 1 gênero";
 	const CAMINHO_PARA_API_STATUS = "/api/anime/status";
-	const MESSAGEM_SUCESSO_CADASTRO = "Sucesso ao cadastrar o anime!";
+	const MENSAGEM_SUCESSO_CADASTRO = "Sucesso ao cadastrar o anime!";
+	const MENSAGEM_SUCESSO_EDITAR = "Sucesso ao editar o anime!"
 	const CAMINHO_PARA_API_ADICIONAR_ANIME = "/api/anime/adicionar";
+	const CAMINHO_PARA_API_EDITAR_ANIME = "/api/anime/atualizar"
+	const CAMINHO_PARA_API_ANIME = "/api/anime/"
 	const OPCAO_VOLTAR_PARA_LISTA_DE_ANIME = "Voltar a lista de anime";
 	const POST = 'POST';
-
-	return ControleBase.extend("ui5.anime.app.anime.CadastroAnime", {
-		onInit: async function () {
+	const PUT = 'PUT';
+	const POSICAO_CADASTRO_OU_EDITAR = 1;
+	const POSICAO_ID_DO_ANIME = 2;
+	const LABEL_ID = 'labelId';
+	const INPUT_ID = 'inputId';
+	const HASH_EDITAR = 'editar';
+	const ID_DA_PAGINA = "pagina";
+	let parametros = '';
+	return ControleBase.extend("ui5.anime.app.anime.CadastroEditarAnime", {
+		onInit: function () {
 			const oRota = this.getOwnerComponent().getRouter();
 			oRota.getRoute(ROTA_ADICIONAR_ANIME).attachMatched(this._aoCoincidirRota, this);
-			
+			oRota.getRoute(ROTA_EDITAR_ANIME).attachMatched(this._aoCoincidirRota, this);
+		},
+		_aoCoincidirRota: function () {
+			this._exibirEspera(async () => {
+				this._limparCampos();
+				parametros = this._getRota().getHashChanger().getHash().split('/');
+				this._modelo(await HttpRequest._request(CAMINHO_PARA_API_GENEROS), NOME_DO_MODELO_LISTA_DE_GENEROS);
+				this._modelo(await HttpRequest._request(CAMINHO_PARA_API_STATUS), NOME_DO_MODELO_DA_LISTA_DE_STATUS);
+				if (parametros[POSICAO_CADASTRO_OU_EDITAR] == HASH_EDITAR) {
+					this.byId(ID_DA_PAGINA).setTitle(TITULO_EDITAR);
+					this.byId(LABEL_ID).setVisible(true);
+					this.byId(INPUT_ID).setVisible(true);
+					this._definirDados();
+				}
+				else{
+					this.byId(ID_DA_PAGINA).setTitle(TITULO_CADASTRO);
+					this.byId(LABEL_ID).setVisible(false);
+					this.byId(INPUT_ID).setVisible(false);
+				}
+			})
+		},
+		_definirDados: async function () {
+			var anime = await HttpRequest._request(CAMINHO_PARA_API_ANIME + parametros[POSICAO_ID_DO_ANIME]);
+			this.byId(INPUT_ID).setValue(anime.id);
```

Comment:
quando for definir os dados na tela utiliza o modelo, eu sei que assim funciona mas o padrao da empresa nao utiliza